pyccel/codegen/pipeline.py

```diff
@@ -334,6 +334,12 @@ def handle_error(stage):
 
             mod_obj.add_dependencies(stdlib)
 
+
+    # Iterate over the external_libs list and determine if the printer
+    # requires an external lib to be included.
```

Comment:
Since STC only depends on header files and doesn't require a compiled object, we can use a map with the library name as the key and its path as the value, this way we can avoid static lookups to retrieve the library name and its path. The same method can be applied to future external libraries.
Please check this update in commit `98967292141cfa853b1b8ea5c4dec35334210dc9` and share your opinion.